src/lib.rs

```diff
@@ -22,61 +22,45 @@
 //!
 //! # Examples
 //!
-//! Usage examples:
-//!
 //! * [`CellBuffer` example](crate::CellBuffer#example)
 //! * [`MaskedCellBuffer` example](crate::MaskedCellBuffer#example)
+//! * [GDAL example](crate::RasterBandEx::read_cells) (_requires `gdal` feature flag_)
 //!
 //! # Feature Flags
 //!
 //! The following feature flags are available.
 //!
-//! | Name     | Description                          | Default |
-//! |:--------:|--------------------------------------|:-------:|
-//! | `masked` | Enable the `MaskedCellBuffer` API    | `true`  |
-//! | `serde`  | Derive `serde` traits for core types | `true`  |
+//! | Name     | Description                                    | Default |
+//! |:--------:|------------------------------------------------|:-------:|
+//! | `masked` | Enable the `MaskedCellBuffer` API              | `true`  |
+//! | `serde`  | Derive `serde` traits for core types           | `true`  |
+//! | `gdal`   | Enable `CellBuffer`s in `georust/gdal` API[^1] | `false` |
 //!
+//! [^1]: Note: `gdal` disables cell-types `UInt64`, `Int8`, and `Int64`
+//! to be backwards compatible with older versions of GDAL.
 
 mod buffer;
 mod ctype;
 mod encoding;
 pub mod error;
+#[cfg(feature = "gdal")]
+mod gdal;
 #[cfg(feature = "masked")]
 mod masked;
 mod value;
 
 pub use buffer::*;
 pub use ctype::*;
 pub use encoding::*;
+#[cfg(feature = "gdal")]
+pub use gdal::*;
 #[cfg(feature = "masked")]
 pub use masked::*;
 use std::fmt::{Debug, Formatter};
-pub use value::ops::*;
 pub use value::*;
 
-/// A [callback style](https://danielkeep.github.io/tlborm/book/pat-callbacks.html)
-/// macro used to construct various implementations covering all [`CellType`]s.
-///
-/// It calls the passed identifier as a macro with two parameters:
-/// * the cell type id (e.g. `UInt8`),
-/// * the cell type primitive (e.g. `u8`).
-///
-/// # Example
-/// ```rust
-/// use erased_cells::{with_ct, CellType};
-/// fn primitive_name(ct: CellType) -> &'static str {
-///     macro_rules! primitive_name {
-///        ($(($id:ident, $p:ident)),*) => {
-///             match ct {
-///                 $(CellType::$id => stringify!($p),)*
-///             }
-///        };
-///     }
-///     with_ct!(primitive_name)
-/// }
-///
-/// assert_eq!(primitive_name(CellType::Float32), "f32");
-/// ```
+#[cfg(not(feature = "gdal"))]
+#[doc=include_str!("with_ct.md")]
```

Comment:
I moved the macro documentation into a separate file so I didn't have to have to copy it  in both `gdal` and !`gdal` forms.

Reply:
OBE